pkg/controllers/provisioner/v1alpha1/allocation/greedyallocator.go

```diff
@@ -41,13 +44,79 @@ func (a *GreedyAllocator) Allocate(provisioner *v1alpha1.Provisioner, pods []*v1
 	zap.S().Infof("Allocating %d pending pods from %d constraint groups", len(pods), len(groups))
 	// 2. Group pods into equally schedulable constraint group
 	for _, group := range groups {
-		if _, err := a.CloudProvider.CapacityFor(&provisioner.Spec).Create(context.TODO(), group.Constraints); err != nil {
+		nodes, err := a.CloudProvider.CapacityFor(&provisioner.Spec).Create(context.TODO(), group.Constraints)
+		if err != nil {
 			return fmt.Errorf("while creating capacity, %w", err)
 		}
+		if err := a.createNodesAndAssignPods(nodes, group.Pods); err != nil {
+			return fmt.Errorf("assigning pods to nodes err: %w", err)
+		}
+	}
+	return nil
+}
+
+func (a *GreedyAllocator) createNodesAndAssignPods(nodes []*v1.Node, pods []*v1.Pod) error {
```

Comment:
As discussed, I think most of this logic needs to happen in the cloud provider. Will resolve in a followon